src/register.php

```diff
@@ -33,34 +34,23 @@
 			// TODO: remove this column from the database
 			$IP = 0;
 
-			$sql = "SELECT email FROM pending_validations WHERE email=?";
-			$rows = [];
-			if ( $sth = $dice->dbconn->prepare( $sql )) {
-				$sth->bind_param('s',$email);
-				$sth->execute() or trigger_error($mysqli->error);
-				$sth->bind_result($emailColumn);
-				while($sth->fetch()) {
-					$rows[] = $emailColumn;
-				}
-			} else {
-				echo "A DB error has occured, please contact an admin. (";
-				var_dump( $dice->dbconn->errno );
-				echo ")";
-				exit;
-			}
+			$sth = $dice->dbconn->prepare("SELECT email FROM pending_validations WHERE email=?");
+			$sth->bind_param('s', $email);
+			$sth->execute() or trigger_error($mysqli->error);
 
 			// insert or update pending validation
-			if (empty($rows)) {
+			if ($sth->num_rows) {
```

Comment:
Should this be `if ($sth->num_rows === 0) {`?

Reply:
Yes, my bad